packages/core/src/command/consolidated/consolidated-filter-by-date.ts

```diff
@@ -0,0 +1,334 @@
+    const areDates = areDatesWithinRange(dates, range);
+    if (areDates !== undefined) return areDates;
+  }
+  if (resource.date) {
+    const res = areDatesWithinRange([resource.date], range);
+    if (res !== undefined) return res;
+  }
+  return includeIfNoDate;
+}
+
+function isDiagnosticReportWithinDateRange(resource: DiagnosticReport, range: DateRange): boolean {
+  const dates = getDatesFromEffectiveDateTimeOrPeriod(resource);
+  const res = areDatesWithinRange(dates, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isDocumentReferenceWithinDateRange(
+  resource: DocumentReference,
+  range: DateRange
+): boolean {
+  const res = isPeriodWithinRange(resource.context?.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isConsentWithinDateRange(resource: Consent, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.dateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isCoverageWithinDateRange(resource: Coverage, range: DateRange): boolean {
+  const res = isPeriodWithinRange(resource.period, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isGoalWithinDateRange(resource: Goal, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.startDate, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isImmunizationWithinDateRange(resource: Immunization, range: DateRange): boolean {
+  const res = isDateWithinDateRange(resource.occurrenceDateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isMedicationDispenseWithinDateRange(
+  resource: MedicationDispense,
+  range: DateRange
+): boolean {
+  const res = isDateWithinDateRange(resource.whenHandedOver, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isMedicationRequestWithinDateRange(
+  resource: MedicationRequest,
+  range: DateRange
+): boolean {
+  const res = isDateWithinDateRange(resource.authoredOn, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isMedicationAdministrationWithinDateRange(
+  resource: MedicationAdministration,
+  range: DateRange
+): boolean {
+  const res = isDateWithinDateRange(resource.effectiveDateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isMedicationStatementWithinDateRange(
+  resource: MedicationStatement,
+  range: DateRange
+): boolean {
+  const res = isDateWithinDateRange(resource.effectiveDateTime, range);
+  if (res !== undefined) return res;
+  return includeIfNoDate;
+}
+
+function isObservationWithinDateRange(resource: Observation, range: DateRange): boolean {
+  if (resource.effectiveDateTime) {
+    const res = isDateWithinDateRange(resource.effectiveDateTime, range);
```

Comment:
we have effectivePeriod in Observation that we use also

Reply:
good catch, totally missed that 😬